### DISCUSSION TITLE
Issue with API routes and AWS SES

### DISCUSSION THREAD

**Discussion by adherb** [vercel/vercel#8335] [Help] [answered]
Hello,

I am having an issue with using [this package](https://www.npmjs.com/package/@aws-sdk/client-ses), to send emails using AWS SES and Next.js API routes.

When checking the API logs I'm receiving a authorisation error, where basically the package is trying to use Vercel's AWS ID, rather my AWS ID. The strange thing is locally on my computer it works perfectly. 

Here is what I'm doing

```
// /api/sendEmail

import { SESClient, SendEmailCommand } from "@aws-sdk/client-ses";

const client = new SESClient({ region: "us-east-1" });

let params = {
      Destination: {
        ToAddresses: usersEmail,
      },
      Message: {
        Body: {
          Html: {
            Charset: "UTF-8",
            Data: `<p>{message}</p>
    `,
          },
          Text: {
            Charset: "UTF-8",
            Data: `{message}`,
          },
        },
        Subject: {
          Charset: "UTF-8",
          Data: `Email Subject`,
        },
      },
      Source: process.env.SUPPORT_EMAIL,
      ReplyToAddresses: [process.env.SUPPORT_EMAIL],
    };

await client.send(new SendEmailCommand(params));
```

As you can see a super simple piece of node, with no issues when running locally.

I won't post the error from the functions log, as I don't want to expose AWS account ID that are not mine. But I have double checked with an AWS engineer and the AWS account ID  that is being used is not mine.

Does anyone know how I can get round this? Should I add more params when I initiate the client in order to get AWS to use my account.

**Comment by leerob** [comment, marked as answer]
> The package is trying to use Vercel's AWS ID, rather my AWS ID. The strange thing is locally on my computer it works perfectly.

This is because there are [reserved environment variables](https://vercel.com/docs/concepts/projects/environment-variables#reserved-environment-variables) on Vercel which are pointing to Vercel's AWS environment instead of your own. On your local machine, it's using your local environment variables (likely set when you were using the AWS CLI).

The solution is to add the proper [Environment Variables on Vercel](https://vercel.com/docs/concepts/projects/environment-variables) for your AWS account and credentials 😄

**Reply by adherb** [reply]
Thanks for you fast reply @leerob, appreciated as always! Makes perfect sense, as I haven't specified them in the params they must be using Vercels.
